pkg/apis/resource/v1alpha1/storage/gcs.go

```diff
@@ -1,172 +0,0 @@
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-package storage
-
-import (
-	"fmt"
-	"path/filepath"
-	"strings"
-
-	"github.com/tektoncd/pipeline/pkg/apis/pipeline"
-	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
-	resourcev1alpha1 "github.com/tektoncd/pipeline/pkg/apis/resource/v1alpha1"
-	"github.com/tektoncd/pipeline/pkg/names"
-	corev1 "k8s.io/api/core/v1"
-)
-
-const (
-	gcsSecretVolumeMountPath     = "/var/secret"
-	activateServiceAccountScript = `#!/usr/bin/env bash
-if [[ "${GOOGLE_APPLICATION_CREDENTIALS}" != "" ]]; then
-  echo GOOGLE_APPLICATION_CREDENTIALS is set, activating Service Account...
-  gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS}
-fi
-`
-)
-
-// GCSResource is a GCS endpoint from which to get artifacts which is required
-// by a Build/Task for context (e.g. a archive from which to build an image).
-type GCSResource struct {
-	Name     string                                `json:"name"`
-	Type     resourcev1alpha1.PipelineResourceType `json:"type"`
-	Location string                                `json:"location"`
-	TypeDir  bool                                  `json:"typeDir"`
-	// Secret holds a struct to indicate a field name and corresponding secret name to populate it
-	Secrets []resourcev1alpha1.SecretParam `json:"secrets"`
-
-	ShellImage  string `json:"-"`
-	GsutilImage string `json:"-"`
-}
-
-// NewGCSResource creates a new GCS resource to pass to a Task
-func NewGCSResource(name string, images pipeline.Images, r *resourcev1alpha1.PipelineResource) (*GCSResource, error) {
-	if r.Spec.Type != resourcev1alpha1.PipelineResourceTypeStorage {
-		return nil, fmt.Errorf("GCSResource: Cannot create a GCS resource from a %s Pipeline Resource", r.Spec.Type)
-	}
-	var location string
-	var locationSpecified, dir bool
-
-	for _, param := range r.Spec.Params {
-		switch {
-		case strings.EqualFold(param.Name, "Location"):
-			location = param.Value
-			if param.Value != "" {
-				locationSpecified = true
-			}
-		case strings.EqualFold(param.Name, "Dir"):
-			dir = true // if dir flag is present then its a dir
-		}
-	}
-
-	if !locationSpecified {
-		return nil, fmt.Errorf("GCSResource: Need Location to be specified in order to create GCS resource %s", r.Name)
-	}
-	return &GCSResource{
-		Name:        name,
-		Type:        r.Spec.Type,
-		Location:    location,
-		TypeDir:     dir,
-		Secrets:     r.Spec.SecretParams,
-		ShellImage:  images.ShellImage,
-		GsutilImage: images.GsutilImage,
```

Comment:
Removing this might also let us remove `GsutilImage`, all the way up to the flag in controller.yaml.

Reply:
Thanks Jason for catching this. Removed 😺